osu.Game/OsuGame.cs

```diff
@@ -284,9 +284,8 @@ public void PresentScore(ScoreInfo score)
             {
                 Ruleset.Value = databasedScoreInfo.Ruleset;
                 Beatmap.Value = BeatmapManager.GetWorkingBeatmap(databasedBeatmap);
-                Mods.Value = databasedScoreInfo.Mods;
 
-                menuScreen.Push(new PlayerLoader(() => new ReplayPlayer(databasedScore)));
+                menuScreen.Push(new ReplayPlayerLoader(() => new ReplayPlayer(databasedScore), databasedScoreInfo.Mods));
```

Comment:
why can't the `ReplayPlayerLoader` make the `ReplayPlayer`?

Reply:
it just seems more consistent to have it take a delegate like the other `PlayerLoader`s, but sure.

Reply:
Why? It would need the same code in every usage.